lib/system/nuttx/io.c

```diff
@@ -75,13 +75,13 @@ static void metal_io_close_(struct metal_io_region *io)
 static metal_phys_addr_t metal_io_offset_to_phys_(struct metal_io_region *io,
 						  unsigned long offset)
 {
-	return up_addrenv_va_to_pa((void *)offset);
+	return up_addrenv_va_to_pa(io->virt + offset);
```

Comment:
pointer void * + offset, this will not work for all compilers. As the size of *(void * p) is not determined.
Better to use char *

Reply:
Fixed